Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm

```diff
@@ -560,8 +568,17 @@ static bool hasValue(const ImageBufferBackendHandle& backendHandle)
     }
 
     IntRect layerBounds(IntPoint(), expandedIntSize(m_parameters.size));
-    if (!m_dirtyRegion.contains(layerBounds) && m_backBuffer.imageBuffer)
-        context.drawImageBuffer(*m_backBuffer.imageBuffer, { 0, 0 }, { CompositeOperator::Copy });
+    if (!m_dirtyRegion.contains(layerBounds) && m_backBuffer.imageBuffer) {
+        if (!previouslyPaintedRect)
+            context.drawImageBuffer(*m_backBuffer.imageBuffer, { 0, 0 }, { CompositeOperator::Copy });
+        else {
+            Region copyRegion(*previouslyPaintedRect);
+            copyRegion.subtract(m_dirtyRegion);
```

Comment:
I wonder if it’s worth doing the Region math here. Maybe just a rect intersection of `dirtyBounds` and `* previouslyPaintedRect` would be enough?

Reply:
Intersection would give the wrong rectangle wouldn't it? We want to copy the pixels that aren't in dirtyBounds, since those will be repainted this frame.

Reply:
Yeah that wasn't right. There's probably an approximation you could do with rects, but there are only a few rects here so the region math should be cheap enough.